src/wolframite/base/package.clj

```diff
@@ -0,0 +1,49 @@
+(ns wolframite.base.package
+  "The place to be for loading and manipulating Wolfram packages.
+"
+  (:require
+   [clojure.string :as string]
+   [wolframite.core :as wl]
+   [wolframite.impl.wolfram-syms.intern :as wi]
+   [wolframite.wolfram :as w]))
+
+(defn load-symbols
+  "Finds all of the public symbols in the given context and interns them to a clojure-accesible namespace. This is put under the given symbol or the context by default.
+
+  N.B. context does not contain the trailing '`'"
+  ([context]
+   (load-symbols context (symbol context)))
+  ([context sym--ns]
+   (let [names (wl/eval (w/Names (str context "`*")))
+         docs (mapv (fn [name]
+                      (mapv (fn [attr] (wl/eval (w/Information name attr)))
+                            ["FullName" "Usage"]))
+                    names)]
+
+     (doall (map (fn [[name-sym doc]]
+                   (wi/clj-intern (symbol (-> (string/split name-sym #"`") last))
+                                  {:intern/ns-sym sym--ns
+                                   :intern/extra-meta {:doc doc}}))
+                 docs)))))
+
+(defn <<
+  "Get a Wolfram package and adds the constants/functions etc. under a clojure-accessible symbol: either the given one or the name of the context by default.
+
+  TODO: Should check that the symbol isn't already being used."
```

Comment:
An example of how this is used with our demo package would be nice here, in the docstring.

Normally, in clj, I'd expect `fname` to be a simple symbol but it seems to be not function-name but file-path. Then I'd prefer to call it that :-) 

Could we settle on either using "package" or "context"? Though it perhaps only seem confusing to me, a newbe who knows 0 about what each means.

`sym` is very generic name that communicates nothing. I assume it is actually `ns-name`, where the namespace does not need to exist?

Reply:
Given the points above (and, being ignorant of Wolfram, I may well be wrong in some/all), I'd suggest

```clj
(defn load-package!
   "..."
   ([path-to-package package-name alias] ...) ...)
```

Notice the `alias` rather than `sym` b/c we are used from Clojure's require to alias namespaces. Though if Wolf. has different terminology then we may want to prefer that over clojure one...

Reply:
Updated.
Unfortunately, package and context are slightly different things (see above). In this case, I'd rather keep the Wolfram terminology for Wolfram things because it makes it easier for people to find the relevant documentation. Similarly, I like alias instead of sym as it's a clojure name for a clojure thing. I added some docs at our side in any case.